examples/topics/geo/eclipse.py

```diff
@@ -4,7 +4,7 @@
 
 .. bokeh-example-metadata::
     :sampledata: us_states
-    :apis: bokeh.plotting.figure.patches, bokeh.models.sources.ColumnDataSource, bokeh.models.mappers.LinearColorMapper, bokeh.models.annotations.ColorBar
+    :apis: bokeh.plotting.figure.patches, bokeh.models.sources.ColumnDataSource, bokeh.models.mappers.LinearColorMapper, bokeh.models.ColorBar
```

Comment:
If we're removing `annotations.` then why keep `sources.` and `mappers.`?

Reply:
I started out with fixing the broken link to `LegendItem` in https://docs.bokeh.org/en/latest/docs/user_guide/basic/annotations.html#hiding-legend-items - I'll look into more similar cases, with a focus on fixing non-functional links in the docs and will keep updating this PR...

Reply:
If I see correctly, something like *:class:`~bokeh.models.annotations.Title.align`* produces a broken link in the docs, while *:class:`~bokeh.models.Title.align`* works correctly.
On the other hand, *:class:`~bokeh.models.sources.ColumnDataSource`* seems to work just fine.

My goal is to reduce the number of broken links as much as possible. But it seems like there is some inconsistency. Would it make more sense to also remove `sources.` and `mappers.`, even though they don't seem to lead to broken links? Or can we change things so that Sphinx directives with `.annotation.` in them forward correctly and not lead to broken links? (CC @bryevdv)

Reply:
Users should really only ever be exposed to `bokeh.models` and not to any of the submodules, is my recollection of the intent (we want the freedom to re-organize the submodules).

> On the other hand, :class:~bokeh.models.sources.ColumnDataSource seems to work just fine.

Seems like an unintended accident.  Does `bokeh.models.ColumnDataSource` not work?

Reply:
Yes, for some reason, both `bokeh.models.ColumnDataSource` and `bokeh.models.sources.ColumnDataSource` seem to produce usable links.
We should be consistent here in the docs, so I'll expand this PR to also cover `sources.` and `mappers.`, in addition to `annotations.`.

Reply:
@bryevdv Just to clarify, in the docs, all :ref:, :class:, etc. for bokeh.models should go directly to the object name, with no submodules? There are also many other examples containing submodules in the docs, like `bokeh.models.callbacks.CustomJS`, `bokeh.models.filters.Filter`, `bokeh.models.widgets.tables.DataTable`, `bokeh.models.tools.Toolbar`, `bokeh.models.glyphs.Rect`, and many more...

Reply:
@tcmetzger we might actually want to sit down for a call sometime. I am not sure I can speak authoritatively on the exact current state of things without digging in a bit more. 

I guess I would put my priorties this way:

* would like for users only ever to see `FooModel` or (at most) `bokeh.models.FooModel` in the published docs
* I don't care at all what the *internal* ReST references look like

The slight complication (or maybe just oddness) is that the refguide *is* currently broken up into subsections that correspond to submodules. We did change the section headers to be "plain words" like "Axes" instead of submodule names like `bokeh.models.axes`. But maybe it would really be better  if we just move the ref to be one page per model with no sub-sections. 

That's a larger discussion. But for now I'd say the two priorities above are the guiding principles (assuming you agree with them)